src/darwin/Framework/CHIP/MTRCommissioningParameters.mm

```diff
@@ -47,10 +51,18 @@ - (void)setFailSafeExpiryTimeoutSecs:(NSNumber * _Nullable)failSafeExpiryTimeout
 
 - (NSString *)description
 {
-    // SSID is not required to be UTF-8, but almost always is.
     NSString * ssidString;
     if (self.wifiSSID) {
-        ssidString = [[NSString alloc] initWithData:self.wifiSSID encoding:NSUTF8StringEncoding];
+        // We want to log the SSID, but hash it, so that the actual SSID cannot be
+        // recovered from the log.
+        uint8_t hashedValue[CC_SHA256_DIGEST_LENGTH];
+        CC_SHA256(self.wifiSSID.bytes, static_cast<CC_LONG>(self.wifiSSID.length), hashedValue);
+
+        char hexValue[sizeof(hashedValue) * 2];
+        chip::Encoding::BytesToHex(hashedValue, sizeof(hashedValue), hexValue, sizeof(hexValue), chip::Encoding::HexFlags::kUppercase);
+        ssidString = [[NSString alloc] initWithBytes:hexValue
+                                              length:sizeof(hexValue)
+                                            encoding:NSUTF8StringEncoding];
```

Comment:
Why not just [string hash] per discussion? This is awfully complicated/expensive (not that the cost matters), what's the value here?

Reply:
As I said in the discussion if you do `[string hash]` then to check whether a given SSID is the one that went through this code you have to run `[string hash]` on the _exact same operating system version_ of the same operating system as the one that created the log, no?

Reply:
Yup, which is fine.